themes/qgrid/source/js/menu.js

```diff
@@ -15,41 +15,247 @@ function toggleVisibility(e) {
+				tag.setAttribute('href', href + search);
+			}
+			a.setAttribute('href', href + search);
+		}
+	}
+}
+
+function pageTextSearch(search) {
+	const h2 = document.getElementsByTagName('h2');
+	const p = document.getElementsByTagName('p');
+	for (let searchTarget of [...h2, ...p]) {
+		const a = searchTarget.getElementsByTagName('a')[0];
+		if (a && a.textContent) {
+			searchTarget = a;
+		}
+		highlightSymbols(searchTarget, search);
+	}
+	for (let searchTarget of [...h2, ...p]) {
+		const index = searchTarget.textContent.toLowerCase().indexOf(search);
+		if (index > -1 && search != '') {
+			searchTarget.scrollIntoView();
+			return;
+		}
+	}
+}
+
+function tagSearch(search) {
+	const active = document.getElementsByClassName('active-topic')[0];
+	const nav = document.getElementById('nav');
+	const menuItems = nav.getElementsByTagName('li');
+	for (let i = menuItems.length - 1; i >= 0; i--) {
+		const tags = menuItems[i].dataset.tag;
+		if (tags) {
+			const tag = tags.split(',');
+			next: for (let j = 0; j < tag.length; j++) {
+				const indexTag = tag[j].toLowerCase().indexOf(search);
+				const title = menuItems[i].querySelector('.title');
+				if (title) {
+					if (indexTag > -1 && menuItems[i] != active && search != '' && search != ' ') {
+						menuItems[i].style.display = '';
+						title.classList.add('menu-item');
+						if (title.textContent[0] != '/') {
+							title.prepend("/ ");
+						}
+						displayTag(menuItems[i], tag[j], indexTag, search);
+						break next;
+					}
+					else {
+						let dataFound = menuItems[i].getElementsByTagName('data-found')[0];
+						if (dataFound) {
+							dataFound.style.display = 'none';
+						}
+						title.classList.remove('menu-item');
+						if (search == ' ' || search == '') {
+							const font = title.getElementsByTagName('font')[0];
+							if (font) {
+								const a = font.parentNode;
+								while (font.firstChild) {
+									a.insertBefore(font.firstChild, font);
+								}
+							}
+						}
+						title.textContent = title.textContent.replace("/ ", '');
+					}
+				}
 			}
 		}
 	}
 }
 
+function menuItemsSearch(search) {
+	const nav = document.getElementById('nav');
+	const menuItems = nav.getElementsByTagName('li');
+	for (let i = 0; i < menuItems.length; i++) {
+		const title = menuItems[i].querySelector('.title');
+		const dataFound = menuItems[i].getElementsByTagName('data-found')[0];
+		if (title || dataFound) {
+			let indexMenu, indexTag;
+			if (title) {
+				indexMenu = title.textContent.toLowerCase().indexOf(search);
+				highlightSymbols(title, search);
+			}
+			if (dataFound) {
+				const tag = dataFound.getElementsByTagName('a')[0];
+				indexTag = tag.textContent.toLowerCase().indexOf(search);
+				highlightSymbols(tag, search);
+			}
+			if (indexMenu > -1 || indexTag > -1) {
+				menuItems[i].style.display = '';
+			}
+			else {
+				menuItems[i].style.display = 'none';
+			}
+		}
+	}
+}
+
+function displayTag(line, tag, indexTag, search) {
```

Comment:
showTag is better name,

too much arguments in the function, let's limit arguments number to 3. Please suggest how we can do it differently